### PR TITLE
Handle phpstan array shapes

### PR DESCRIPTION
**Pull request by dgafka**
This PR provides possibility to serialize and deserialize phpstan array shapes. 

As no matter of the structure `JMS` can handle arrays, instead of throwing exception when we meet phpstan array shape, we can simply serialize it as array.

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| Doc updated   | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT

### CODE REVIEW

**Comment by dgafka** [comment]
@goetas @scyzoryck wdyt? :)

File: tests/Fixtures/DocBlockType/PhpstanMultipleArrayShapes.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+declare(strict_types=1);
+
+namespace JMS\Serializer\Tests\Fixtures\DocBlockType;
+
+/**
+ * @phpstan-type Settings array{
+ *  method: string,
+ *  amount: array{type: string, value: string|null}
+ * }
+ * @phpstan-type Details array{
```
Comment (scyzoryck) [inline tests/Fixtures/DocBlockType/PhpstanMultipleArrayShapes.php:12]: Can we add also test & support for some types like: `array<int, SomeType>` or `array<SomeType>`, please? It is already supported by Serializer, so I believe it should be also supported for resolving types here.

**Review by scyzoryck — CHANGES_REQUESTED** [review]
Thanks for contribution! I believe it might be useful for users. Could you also update documentation, please? 

Best, scyzoryck.

File: src/Metadata/Driver/DocBlockDriver/DocBlockTypeResolver.php [hunk 2]
```diff
@@ -227,6 +228,17 @@ private function expandClassNameUsingUseStatements(string $typeHint, \Reflection
             }
         }
 
+        if ($declaringClass->getDocComment()) {
+            preg_match_all(self::PHPSTAN_ARRAY_SHAPE, $declaringClass->getDocComment(), $foundPhpstanArrayShape);
+            $phpstanArrayShapes = array_map(static function (string $phpstanArrayType) {
+                return $phpstanArrayType;
```
[the remark this answers is not in the document]
Comment (dgafka) [reply src/Metadata/Driver/DocBlockDriver/DocBlockTypeResolver.php]: code refactored

File: tests/Fixtures/DocBlockType/PhpstanMultipleArrayShapes.php (hunk 1 above)
Comment (dgafka) [reply tests/Fixtures/DocBlockType/PhpstanMultipleArrayShapes.php]: Ok, I've added support array collections  :)

**Comment by dgafka** [comment]
@scyzoryck can you re-review?

**Review by scyzoryck — APPROVED** [review]
Thanks again for contribution! 👍

**Comment by goetas** [comment]
can you please rebase?

**Comment by dgafka** [comment]
@goetas done :)

**Pull request merged**